### ISSUE TITLE
Add popup menu support

### ISSUE THREAD

**Issue by ecton** [khonsulabs/cushy#103]
A popup menu is likely a Grid (after #100) + vertical Scroll widget being shown on an overlay layer.

Showing a popup menu needs to support being placed near an arbitrary location. As of writing this, the overlay widget only supports placing relative to a widget, so this is potentially new functionality for the overlay.

The popup menu needs to support type-to-select. However, I want any widget to be able to be used as a menu item, so this needs to either be callback based or we need a way to specify the type-ahead-string for a given menu item.

The API for a popup menu needs to support being notified if the menu is dismissed without a selection being made. a `OnceCallback` can be used to signal when a selection is made or cancelled.

It might be nice to have a dedicated widget that renders a chevron and displays the associated popup menu. This could then be used within a button's label to offer a push button with an optional popup menu.

**Comment by ecton** [comment]
As of 07f1febe9fbf4514ee07f1d92eac8ccb584de307 this is mostly implemented. Opening new issues to track remaining features.


https://github.com/khonsulabs/cushy/assets/50053/264b15cf-be01-4f0e-848e-cae574029596
